### PR TITLE
Fitting 8 known CNS signatures to data, 3/n (3/3?)

### PR DESCRIPTION
**Pull request by sjspielman**
#### What scientific question is your analysis addressing?

Next stage in mutational signature analysis.

#### What was your approach?

Use `sigfit` to infer exposures to 8 known (adult) CNS signatures. Used 3000 iterations with the poisson model (shows virtually identical performance to NMF but is much faster). _This analysis needed 3-4 GB RAM and takes about an hour. Pretty reasonable!_

#### What GitHub issue does your pull request address?

Issue #818 

#### Which areas should receive a particularly close look?

+ I had to do some manually CSV downloading to grab the signatures in the first place, and then write a quick script to merge everything into an RDS file for use by `sigfit`. All of that is in `util/refsig_cns_data/` with documentation, but it is _not_ currently run with the rest of the analysis pipeline. **Should it be?**
+ Visualization is currently with a _barplot of mean/median mutation counts_, colored with the histology palette. Since we are just fitting known signatures to samples, _all_ samples have a non-0 burden, and therefore bubble plots used in other circumstances aren't as interesting here (bubbles are usually sizes by % of samples containing signature, but everything here is "1"). 
  + Plot very much subject to edits, initial draft to get the concept down. Will make pub ready if we like the concept.

#### Is there anything that you want to discuss further?

+ What other plots would we like to see here? Barplots filled by different attributes? Barplots filled with signature and use a different variable on the x-axis? Can also look at individual _samples_, for example plot exposures to each signature across the "top 10 most mutated samples"?

+ Do we think this PR will wrap up mutational signatures, or would additional inferences be good to do?

+ Once this PR gets more towards the merging stage, I'll fill in the analysis README and add all results/plots to project documentation

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

A _little_ review :)

#### What types of results are included (e.g., table, figure)?

Results file - RDS containing exposures across samples: `fitted_cns_signature_exposures.RDS`
Plot - `plots/mean_median_exposure_barplot.png`

#### What is your summary of the results?

+ Mean/median counts are very different, presumably from variation in mutation burden across samples in PBTA dataset
+ Results are mostly descriptive

#### Reproducibility Checklist

- [X] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [X] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [X] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by sjspielman** [comment]
Looks like we'll be good with loading the `signatures.tools.libs` library to obtain CNS signatures, but the values to do match up with CNS signatures released on the website. We will use values in the package with 8 signatures named A-H. 

I'd like to add a barplot showing mutation counts for the top-10 most mutated samples, and bars would be filled by *signature grouping.* However, but I'm not seeing an option for *discrete* color palettes [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/figures). @jaclyn-taroni any recommendations?

File: Dockerfile
Comment (jaclyn-taroni) [inline Dockerfile:404]: Let's tie to a specific commit
```suggestion
RUN R -e "remotes::install_github('Nik-Zainal-Group/signature.tools.lib', ref = '73e899c9090a215a76a307480bda76c241a4a489')"
```

File: analyses/mutational-signatures/run_mutational_signatures.sh
Comment (jaclyn-taroni) [inline analyses/mutational-signatures/run_mutational_signatures.sh:42]: This is very strange?

File: analyses/mutational-signatures/05-fit_cns_signatures.R
Comment (jaclyn-taroni) [inline analyses/mutational-signatures/05-fit_cns_signatures.R:10]: I might suggest `action = "store_true"` (with `type = "logical"`) here, as that is kind of what I would expect for something called `--abbreviated`, but this opinion is not very strongly held.

File: analyses/mutational-signatures/06-analyze_cns_fit.Rmd
Comment (jaclyn-taroni) [inline analyses/mutational-signatures/06-analyze_cns_fit.Rmd:174]: Can we add a `sessionInfo()` here please?

File: analyses/mutational-signatures/06-analyze_cns_fit.Rmd
Comment (jaclyn-taroni) [inline analyses/mutational-signatures/06-analyze_cns_fit.Rmd:160]: This seems fine for now and perhaps for this purpose more generally.

**Review by jaclyn-taroni — COMMENTED** [review]
This looks pretty good overall, just a couple odds and ends to be resolved before approval & merging.

**Comment by jaclyn-taroni** [comment]
Apparently I had some old comments that were queued but couldn't see on the diff because they're already outdated. Sorry about that!

File: analyses/mutational-signatures/05-fit_cns_signatures.R
Comment (jaclyn-taroni) [reply analyses/mutational-signatures/05-fit_cns_signatures.R]: Okay this says outdated but the point about using `rprojroot` does still stand.

File: analyses/mutational-signatures/run_mutational_signatures.sh
Comment (sjspielman) [reply analyses/mutational-signatures/run_mutational_signatures.sh:42]: What is strange? I'm not seeing it...

**Comment by sjspielman** [comment]
@jaclyn-taroni I'm going to go ahead and fill in all the documentation we were holding off on until now, too.

File: analyses/mutational-signatures/run_mutational_signatures.sh
Comment (jaclyn-taroni) [reply analyses/mutational-signatures/run_mutational_signatures.sh:42]: There's a bunch of whitespace but the little ⛔ indicates there is no new line at the end.

Comment (sjspielman) [reply analyses/mutational-signatures/run_mutational_signatures.sh:42]: Oh I see, that IS weird..... Hmm.  I'll poke around.

**Comment by sjspielman** [comment]
@jaclyn-taroni One question here - should aspects of this pipeline to generate figures in this PR be added into `generate_figures.sh`? Not super clear to me which figures go into that script or not, I guess depends on MS?

**Comment by jaclyn-taroni** [comment]
My general answer is that we should replace the mutational signature figures that are already in `figures` with what you’ve generated with the CNS signatures. I think the _specifics_ (e.g., what figure number, which panel) are very likely to change soon and I don’t have a good idea about how much yet. So that’s to say — I’m not sure if it would be helpful to get those changes started before or after we work out those details to be honest.

**Comment by sjspielman** [comment]
Ok, makes sense. I think that should wait for another PR once the overall figure scheme is more sorted out.

I've updated the README's but didn't have quite enough context to flesh out some of the missing details from the original analysis in `01-known_signatures.Rmd`.

File: analyses/mutational-signatures/05-fit_cns_signatures.R
Comment (sjspielman) [reply analyses/mutational-signatures/05-fit_cns_signatures.R]: I ended up keeping this as `integer` since it seems to play poorly with the CI parameter and I didn't fight the fight with type coercion from character too heavily.

**Review by jaclyn-taroni — APPROVED** [review]
👍🏻

**Pull request merged**
